lib/merchant_repository.rb

```diff
@@ -57,6 +60,10 @@ def delete(id)
     all.delete(find_by_id(id))
   end
 
+  def items_by_merchant_id(id)
+    @engine.items.find_all_by_merchant_id(id)
```

Comment:
Without thinking thru the code as much as you two did building this, I think we could remove this demeter violation by giving SalesEngine it's own method for `items_by_merchant_id`.

Doing so might also make the failing test mentioned in the PR notes easier to write!

I'll pick up here when I start working on this!